test/assume-role.bats

```diff
@@ -86,9 +116,33 @@ teardown() {
   [[ "${lines[23]}" == *"--external-id 123456789012"* ]] || false
 }
 
+@test "should work for the SAML auth scheme" {
+  export AWS_ASSUME_ROLE_AUTH_SCHEME=saml
+  export SAML_IDP_NAME=saml-test-idp
+  run ./assume-role dev look_around us-east-1
+
+  [ "$status" -eq 0 ]
+
+  [ "${lines[0]}" = 'echo "Gathering SAML credentials...";' ]
+  [ "${lines[1]}" = 'echo "Authenticating with SAML provider...";' ]
+  [ "${lines[2]}" = 'echo "Success! IAM session envars are exported.";' ]
+  [ "${lines[3]}" = 'export AWS_REGION="us-east-1";' ]
+  [ "${lines[4]}" = 'export AWS_DEFAULT_REGION="us-east-1";' ]
+  [ "${lines[5]}" = 'export AWS_ACCESS_KEY_ID="role_key_id";' ]
+  [ "${lines[6]}" = 'export AWS_SECRET_ACCESS_KEY="role_secret_key";' ]
+  [ "${lines[7]}" = 'export AWS_SESSION_TOKEN="role_session_token";' ]
+  [ "${lines[8]}" = 'export AWS_ACCOUNT_ID="123456789012";' ]
+  [ "${lines[9]}" = 'export AWS_ACCOUNT_NAME="dev";' ]
+  [ "${lines[10]}" = 'export AWS_ACCOUNT_ROLE="look_around";' ]
+  [ "${lines[19]}" = 'AWS_CONFIG_REGION="nz-north-1";' ]
+  [[ "${lines[25]}" == *"--role-arn arn:aws:iam::123456789012:role/look_around"* ]] || false
+  [[ "${lines[25]}" == *"--principal-arn arn:aws:iam::123456789012:saml-provider/saml-test-idp"* ]] || false
+  [[ "${lines[25]}" = *"--saml-assertion aGVsbG8= --duration-seconds 3600"* ]] || false
```

Comment:
Food for thought: since you seem to be inspecting the first 10 lines, you could consider writing:

```bash
diff <(head -n11 <<< "$lines") <(cat <<EOF
echo "Gathering SAML credentials..."
...etc...
EOF)
```

In this case, if the output matches, `diff succeeds`. If it doesn't, it fails _and_ prints out what changed. Not a blocker, just sharing as an alternative. You'll still need line-specific checks as you have later on.

Reply:
Nice! That's a good idea, might make a future PR to update that.